applications/tari_base_node/src/command_handler.rs

```diff
@@ -535,6 +544,42 @@ impl CommandHandler {
         });
     }
 
+    pub fn ping_peer(&self, dest_node_id: NodeId) {
+        let mut liveness = self.liveness.clone();
+
+        self.executor.spawn(time::timeout(Duration::from_secs(30), async move {
```

Comment:
Maybe just a message if it hits the timeout?

Reply:
Yeah first prize would be to improve liveness service to allow for a ping/pong result to be returned to a caller - for now, I think a message here will work - will add now